lib/template-dependencies.js

```diff
@@ -8,7 +8,7 @@ const crypto = require('crypto');
 const fs = require('fs-extra');
 const spawnAsync = require('cross-spawn-promise');
 const ErrorMessage = require('./error-message');
-const elmJsonCLI = require('elm-json').paths['elm-json'];
+const getExecutable = require('elm-tooling/getExecutable');
```

Comment:
My personal advice would have been to expose `getExecutable` and all other utility functions in the `index.js` file of `elm-tooling`. If you put everything you want to expose as part of the index, then your API becomes much clearer.

If you allow importing from `elm-tooling/xyz`, then people will start doing that and the line between public and private API will become a lot more blurry.

Reply:
`require("elm-tooling")` already means something else – it lets you easily run the `elm-tooling` CLI from a script: https://github.com/lydell/elm-tooling.json/tree/main/cli#elmtoolingcli

`require('elm-tooling/getExecutable')` is a bit “hidden” on purpose. Not many people will use that. In fact, I suspect nobody except elm-review and elm-test will use it. One thing I like with it being a separate entrypoint is that no unnecessary things are `require`:d.

I’ve got this in package.json:

```json
  "exports": {
    ".": "./index.js",
    "./getExecutable": "./getExecutable.js"
  },
```

So the public API is clearly defined, and newer Node.js versions won’t let people import anything else.

Reply:
Oh man, I didn't follow this improvement in Node.js, great! (I should stop burying my head in Elm-land)